dspace-server-webapp/src/main/java/org/dspace/app/rest/security/StatelessAuthenticationFilter.java

```diff
@@ -58,16 +76,34 @@ protected void doFilterInternal(HttpServletRequest req,
                                     HttpServletResponse res,
                                     FilterChain chain) throws IOException, ServletException {
 
-        Authentication authentication = getAuthentication(req);
+        Pair<Authentication, Boolean> pair = getAuthentication(req, res);
+        Authentication authentication = pair.getLeft();
         if (authentication != null) {
             SecurityContextHolder.getContext().setAuthentication(authentication);
             restAuthenticationService.invalidateAuthenticationCookie(res);
         }
-
+        if (pair.getRight()) {
+            return;
+        }
```

Comment:
This entire `Pair` logic seems really odd to me.  Why not just have `getAuthentication()` return either `Authentication` or null?  If an error needs to occur, then `getAuthentication()` should throw an exception, which then gets caught in this method and turned into a `response.sendError()`.

I don't see a scenario where you'd ever want to return *both* an `Authentication` and an error (which is what this `Pair` logic seems to imply could occur).  So, there's no need to have this `Pair`, as either you have an Authentication, it's null, or you throw an error.  Those can all be handled just via normal error handling scenarios.